internal/cluster/transition.go

```diff
@@ -73,6 +73,45 @@ type TransitionHandler interface {
 	SoftTimeoutsEnabled(_ stateswitch.StateSwitch, _ stateswitch.TransitionArgs) (bool, error)
 	FinalizingStageTimeoutMinutes(sCluster *stateCluster) interface{}
 	InstallationTimeoutMinutes(_ *stateCluster) interface{}
+
+	EventHandler
+}
+
+////////////////////////////////////////////////////////////////////////////
+// Event Handler
+////////////////////////////////////////////////////////////////////////////
+
+type EventHandler interface {
+	SendErrorEvent(template string, formatFuncs ...formatFunction) stateswitch.PostTransition
+}
+
+type ErrorEventArgs interface {
+	GetContext() context.Context
+}
+
+func (th *transitionHandler) SendErrorEvent(template string, formatFuncs ...formatFunction) stateswitch.PostTransition {
```

Comment:
I would argue that this function should be called `SendClusterInstallationErrorEvent` or even `SendClusterInstallationFailedEvent` as 

* this is not some abstract event sender that may apply to any kind of event, this is specific to Cluster Installation
* ClusterInstallation Failure is not the only type of failure event handled in this file.

so `SendErrorEvent` does not unambiguously represent what this function does.